internal/ls/userpreferences.go

```diff
@@ -199,7 +309,35 @@ const (
 	QuotePreferenceSingle  QuotePreference = "single"
 )
 
-func (p *UserPreferences) Parse(config map[string]interface{}) {
+func parseQuotePreference(val any) QuotePreference {
+	if s, ok := val.(string); ok {
+		switch strings.ToLower(s) {
+		case "auto":
+			return QuotePreferenceAuto
+		case "double":
+			return QuotePreferenceDouble
+		case "single":
+			return QuotePreferenceSingle
+		}
+	}
+	return QuotePreferenceUnknown
+}
+
+func (p *UserPreferences) Copy() *UserPreferences {
+	if p == nil {
+		return nil
+	}
+	prefCopy := *p
+	copy(prefCopy.AutoImportSpecifierExcludeRegexes, p.AutoImportSpecifierExcludeRegexes)
+	copy(prefCopy.AutoImportFileExcludePatterns, p.AutoImportFileExcludePatterns)
```

Comment:
This kind of copy won't work; prefCopy is a shalow copy, so both args are the same.

Probably you meant `slices.Clone`?